rust/src/lib/ifaces/inter_ifaces_controller.rs

```diff
@@ -234,3 +270,25 @@ pub(crate) fn set_ifaces_up_priority(ifaces: &mut Interfaces) -> bool {
     }
     ret
 }
+
+pub(crate) fn find_unknown_type_port<'a>(
+    iface: &'a Interface,
+    cur_ifaces: &Interfaces,
+) -> Vec<&'a str> {
+    let mut ret: Vec<&str> = Vec::new();
+    if let Some(port_names) = iface.ports() {
+        for port_name in port_names {
+            if let Some(port_iface) =
+                cur_ifaces.get_iface(port_name, InterfaceType::Unknown)
+            {
+                if port_iface.iface_type() == InterfaceType::Unknown {
```

Comment:
Is there really a need for this  additional check?

Reply:
Yes. The previous `get_iface()` will search kernel interfaces with the same name, the returned Interface might have valid interface type.